pages/curriculum/[lesson].tsx

```diff
@@ -15,13 +17,26 @@ import _ from 'lodash'
 
 const Challenges: React.FC<QueryDataProps<AppData>> = ({ queryData }) => {
   const { lessons, session, alerts } = queryData
-  const userSubmissions: UserSubmission[] = _.get(session, 'submissions', [])
-  const lessonStatus: LessonStatus[] = _.get(session, 'lessonStatus', [])
   const router = useRouter()
   const currentlessonId = router.query.lesson as string
+  if (!session) {
+    router.push('/login')
+    return <LoadingSpinner />
+  }
+  if (!lessons || !alerts) {
+    return (
+      <Error title="Internal server error" message="Bad data" src="/500.png" />
+    )
+  }
   const currentLesson: Lesson | undefined = lessons.find(
-    (lesson: Lesson) => lesson.id.toString() === currentlessonId
+    (lesson: Lesson) => lesson.id === currentlessonId
   )
+  if (!currentLesson) {
+    return <Error title="404 error" message="Page not found" src="/404.png" />
```

Comment:
404 should be reserved for the page is not found. Not an error.

Reply:
This case happens if user tries to open lessonID which does not exist, something like https://www.c0d3.com/curriculum/100ShouldNotExist. I could redirect him to 404 page but this way he won't be able to check his incorrect url. 
Do you want me to rename Error component for something else?